src/OrleansAzureUtils/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans;
+using Orleans.AzureUtils.Configuration;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+
+namespace OrleansAzureUtils
+{
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Configure DI container to use AzureBasedMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configureOptions"></param>
+        public static IServiceCollection UseAzureMemebershipTable(this IServiceCollection services,
+            Action<AzureMembershipTableOptions> configureOptions)
+        {
+            services.Configure<AzureMembershipTableOptions>(configureOptions);
+            services.AddSingleton<IMembershipTable, AzureBasedMembershipTable>();
+            return services;
+        }
+
+        /// <summary>
+        /// Configure DI container to use AzureBasedMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="config"></param>
+        public static IServiceCollection UseAzureMemebershipTable(this IServiceCollection services, IConfiguration config)
+        {
+            services.Configure<AzureMembershipTableOptions>(config);
+            services.AddSingleton<IMembershipTable, AzureBasedMembershipTable>();
+            return services;
+        }
+
+
+        /// <summary>
+        /// Configure DI to use AzureBasedMemebershipTable, and get its configuration from <see cref="GlobalConfiguration"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection UseAzureMembershipTableFromLegacyConfigurationSupport(
+            this IServiceCollection services)
+        {
+            services.Configure<GlobalConfiguration, AzureMembershipTableOptions>((configuration, options) =>
+            {
+                options.DataConnectionString = configuration.DataConnectionString;
+                options.DeploymentId = configuration.DeploymentId;
+                options.MaxStorageBusyRetries = configuration.MaxStorageBusyRetries;
+            });
+            services.AddSingleton<IMembershipTable, AzureBasedMembershipTable>();
+            return services;
+        }
+
+        private static void Configure<TConfiguration, TOptions>(this IServiceCollection services,
```

Comment:
I think we should make this method in Orleans.dll public or internal and use that from every extension, no need to copy, maybe a name change there to TransformLegacyConfigurationToOptions or something like that would be better?

Reply:
I think we can make it public. It is a simple function to support. 
I don't think make it internal make sense. Since other csproj which uses it will need to set InternalVisiable to, which is what we want to avoid.